cpp/src/arrow/compute/kernels/scalar_string_ascii.cc

```diff
@@ -95,7 +95,7 @@ struct FixedSizeBinaryTransformExecBase {
                           ctx->Allocate(output_width * input_nstrings));
     uint8_t* output_str = values_buffer->mutable_data();
 
-    const uint8_t* input_data = input.GetValues<uint8_t>(1);
+    const uint8_t* input_data = input.GetValues<uint8_t>(1, input.offset * input_width);
```

Comment:
The current `FixedSizeBinaryTransformExecBase` deals with offsets wrongly. I fixed the bug and added a test for `binary_replace_slice` as well.

Reply:
Nice catch, thank you!